src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -985,71 +1093,49 @@ private int constructMergedTable(Table table, List<FeedToMerge> feedsToMerge,
-    private boolean hasDuplicateError(Set<NewGTFSError> errors) {
-        for (NewGTFSError error : errors) {
-            if (error.errorType.equals(NewGTFSErrorType.DUPLICATE_ID)) return true;
+    private MergeStrategy getMergeStrategy(List<FeedToMerge> feedsToMerge) throws IOException {
+        // Iterate over both feeds to collect all trip and service IDs.
+        for (int i = 0; i < feedsToMerge.size(); i++) {
+            FeedToMerge feed = feedsToMerge.get(i);
+            Set<Table> tablesToCheck = Sets.newHashSet(Table.TRIPS, Table.CALENDAR, Table.CALENDAR_DATES);
+            for (Table table : tablesToCheck) {
+                feed.idsForTable.get(table).addAll(getIdsForTable(feed.zipFile, table));
+            }
         }
-        return false;
-    }
-
-    /** Get table-scoped value used for key when remapping references for a particular feed. */
-    private static String getTableScopedValue(Table table, String prefix, String id) {
-        return String.join(":",
-            table.name,
-            prefix,
-            id);
-    }
-
-    /**
-     * Helper class that collects the feed version and its zip file. Note: this class helps with sorting versions to
-     * merge in a list collection.
-     */
-    private class FeedToMerge {
-        public FeedVersion version;
-        public ZipFile zipFile;
-
-        FeedToMerge(FeedVersion version) throws IOException {
-            this.version = version;
-            this.zipFile = new ZipFile(version.retrieveGtfsFile());
+        FeedToMerge futureFeed = feedsToMerge.get(0);
+        FeedToMerge activeFeed = feedsToMerge.get(1);
+        Set<String> activeTripIds = activeFeed.idsForTable.get(Table.TRIPS);
+        Set<String> futureTripIds = futureFeed.idsForTable.get(Table.TRIPS);
+        Set<String> activeServiceIds = new HashSet<>();
+        activeServiceIds.addAll(activeFeed.idsForTable.get(Table.CALENDAR));
+        activeServiceIds.addAll(activeFeed.idsForTable.get(Table.CALENDAR_DATES));
+        Set<String> futureServiceIds = new HashSet<>();
+        futureServiceIds.addAll(futureFeed.idsForTable.get(Table.CALENDAR));
+        futureServiceIds.addAll(futureFeed.idsForTable.get(Table.CALENDAR_DATES));
+        boolean serviceIdsMatch = activeServiceIds.equals(futureServiceIds);
+        intersectingTripIds = Sets.intersection(activeTripIds, futureTripIds);
+        tripsOnlyInActiveFeed = Sets.difference(activeTripIds, futureTripIds);
+        tripsOnlyInFutureFeed = Sets.difference(futureTripIds, activeTripIds);
+        boolean tripIdsMatch = activeTripIds.equals(futureTripIds);
+        if (serviceIdsMatch && tripIdsMatch) {
+            // Effectively this exact match condition means that the future feed will be used as is
+            // (including stops, routes, etc.), the only modification being service date ranges.
+            // This is Condition 2 in the docs.
```

Comment:
Provide a link to "the docs".

Reply:
Comment updated in https://github.com/ibi-group/datatools-server/pull/376/commits/d6e63dc8a097578040a7639ae604c0f5eea2cfdd.